programs/vote/src/vote_state/mod.rs

```diff
@@ -367,6 +385,23 @@ impl VoteState {
         }
     }
 
+    /// Returns if the vote state contains a slot `candidate_slot`
+    pub fn contains_slot(&self, candidate_slot: Slot) -> bool {
+        for my_vote in &self.votes {
+            match my_vote.slot.cmp(&candidate_slot) {
+                Ordering::Equal => {
+                    return true;
+                }
+                Ordering::Greater => {
+                    return false;
+                }
+                Ordering::Less => (),
+            }
```

Comment:
nit: could you use `self.votes.take_while(|slot| slot <= candidate_slot)` or do a binary search?